flink-python/pyflink/pyflink_gateway_server.py

```diff
@@ -290,7 +291,23 @@ def launch_gateway_server_process(env, args):
         classpath = os.pathsep.join(
             [construct_flink_classpath(env), construct_hadoop_classpath(env)])
         if "FLINK_TESTING" in env:
-            download_apache_avro()
+            retry_times = 3
+            status = 0
+            error = None
+            while retry_times > 0 and not status:
+                retry_times -= 1
+                try:
+                    download_apache_avro()
+                    status = 1
+                except CalledProcessError as e:
+                    status = 0
+                    error = e
+                    print("Retry {0} times to download avro.".format(retry_times))
```

Comment:
Is this times of the output opposite?

Reply:
Print both messages should make it a little clearer. I'll change it